Sources/Storage/Disk/Config/DiskConfiguration.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  DiskConfiguration.swift
+//  StatsdClient
+//
+//  Created by Nghia Tran on 10/9/17.
+//  Copyright © 2017 StatsdClient. All rights reserved.
+//
+
+import Foundation
+
+struct DiskConfiguration: DiskConfigurable {
```

Comment:
```
struct DiskConfiguration: DiskConfigurable {
    static let `default` = DiskConfiguration(name: "default", maxSize: 0, protectionType: something)

    var name: String

    var maxSize: UInt

    var protectionType: FileProtectionType?

}
```
I'd suggest changing `DiskConfiguration` class to something like the above.

What good would a Configuration do if you can't change it's configs?

Reply:
How about using default params in the constructor instead of the static prop?